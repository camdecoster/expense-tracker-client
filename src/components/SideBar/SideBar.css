#SideBar {
    box-sizing: border-box;
    position: fixed;
    /* display: inline-block; */
    height: 100%;
    width: 200px;
    top: 0;
    left: -200px;
    border-right: 1px solid var(--blue-medium);
    transition: 0.5s;
    background-color: inherit;
}

#SideBar a {
    display: block;
    transition: 0.3s;
}

#SideBar.open {
    left: 0;
}

#SideBar .active {
    font-weight: bold;
}
