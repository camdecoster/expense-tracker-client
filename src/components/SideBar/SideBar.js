// React
import React, { Component } from "react";
import { NavLink } from "react-router-dom";

// Configuration
import "./SideBar.css";
// import TrackerContext from "../../contexts/TrackerContext";
import TokenService from "../../services/token-service";

class SideBar extends Component {
    // static contextType = TrackerContext;

    // state = {};

    handleLinkClick = () => {
        // Close user menu
        // this.context.toggleStateBoolean("showNav");
    };

    createNavLinkList = (navLinkTargets) => {
        // Build list of NavLinks for <ul> in SideBar
        let navLinks = "";

        navLinks = navLinkTargets.map((navLink) => {
            return (
                <li key={navLink.link}>
                    <NavLink
                        exact
                        to={"/" + navLink.link}
                        onClick={this.handleLinkClick}
                    >
                        {navLink.name}
                    </NavLink>
                </li>
            );
        });

        return navLinks;
    };

    renderAuthorized = () => {
        const navLinkTargets = [
            {
                name: "Dashboard",
                link: "",
            },
            {
                name: "Add Expense",
                link: "add-expense",
            },
            {
                name: "Expense Log",
                link: "expense-log",
            },
            {
                name: "Categories",
                link: "categories",
            },
            {
                name: "Payment Methods",
                link: "payment-methods",
            },
            // {
            //     name: "Account",
            //     link: "account",
            // },
        ];

        return this.createNavLinkList(navLinkTargets);
    };

    renderUnauthorized = () => {
        const navLinkTargets = [
            {
                name: "Log In",
                link: "login",
            },
            {
                name: "Register",
                link: "register",
            },
        ];

        return this.createNavLinkList(navLinkTargets);
    };

    render() {
        return (
            <div id='SideBar' className={this.props.className}>
                <h3>Expense Tracker</h3>

                {/* Links and logout should only be shown if user logged in */}
                <ul className='link_list'>
                    {TokenService.hasAuthToken()
                        ? this.renderAuthorized()
                        : this.renderUnauthorized()}
                </ul>
            </div>
        );
    }
}

export default SideBar;
