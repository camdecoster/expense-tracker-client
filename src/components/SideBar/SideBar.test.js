// React
import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

// Configuration
// import TrackerContext from "../../contexts/TrackerContext";
import TokenService from "../../services/token-service";

// Components
import SideBar from "./SideBar";

// Create mock service for TokenService
jest.mock("../../services/token-service");

describe("<SideBar> component", () => {
    // Create dummy context values for test
    // const contextValue = {
    //     showUserMenu: false,
    // };

    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <MemoryRouter>
                <SideBar />
            </MemoryRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    describe("renders the UI", () => {
        it("render the unauthorized UI", () => {
            TokenService.hasAuthToken.mockReturnValue(false);

            const tree = renderer
                .create(
                    <MemoryRouter>
                        <SideBar />
                    </MemoryRouter>
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it("renders the authorized UI", () => {
            TokenService.hasAuthToken.mockReturnValue(true);

            const tree = renderer
                .create(
                    <MemoryRouter>
                        <SideBar />
                    </MemoryRouter>
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
