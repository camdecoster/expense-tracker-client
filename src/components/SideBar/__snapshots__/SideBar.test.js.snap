// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar> component renders the UI render the unauthorized UI 1`] = `
<div
  id="SideBar"
>
  <h3>
    Expense Tracker
  </h3>
  <ul
    className="link_list"
  >
    <li>
      <a
        aria-current={null}
        href="/login"
        onClick={[Function]}
      >
        Log In
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
  </ul>
</div>
`;

exports[`<SideBar> component renders the UI renders the authorized UI 1`] = `
<div
  id="SideBar"
>
  <h3>
    Expense Tracker
  </h3>
  <ul
    className="link_list"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Dashboard
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/expenses/new"
        onClick={[Function]}
      >
        New Expense
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/expenses"
        onClick={[Function]}
      >
        Expenses
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/categories"
        onClick={[Function]}
      >
        Categories
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/payment-methods"
        onClick={[Function]}
      >
        Payment Methods
      </a>
    </li>
  </ul>
</div>
`;
